### ISSUE TITLE
`SessionCookieConfig` support causes session cookie to immediately expire on Undertow

### ISSUE THREAD

**Issue by vpavic** [spring-projects/spring-session#805]
See [#87 (comment)](https://github.com/spring-projects/spring-session/issues/87#issuecomment-309431537).

**Comment by stevestorey** [comment]
For anyone else ending up here, there is a fairly simple workaround, which  is just to define your own serializer as a bean in the configuration on which you're applying `@EnableSpringHttpSession` / `@EnableRedisHttpSession` / etc.
```
@Bean
CookieSerializer cookieSerializer() {
    return new DefaultCookieSerializer();
}
```

**Comment by vpavic** [comment]
This is caused by Undertow not respecting `javax.servlet.SessionCookieConfig#getMaxAge` contract which states that by default `-1` should be returned:

> Gets the lifetime (in seconds) of the session tracking cookies created on behalf of the application represented by the `ServletContext` from which this `SessionCookieConfig` was acquired.
By default, `-1` is returned.

[Source](https://docs.oracle.com/javaee/7/api/javax/servlet/SessionCookieConfig.html#getMaxAge--)

I've opened [UNDERTOW-1111](https://issues.jboss.org/browse/UNDERTOW-1111).

@rwinch I guess we should try to detect Undertow and avoid applying our `SessionCookieConfig` integration in that case. WDYT?

**Comment by rwinch** [comment]
@vpavic Thanks for looking into this. Given it is a bug in Undertow and there is a simple workaround, I think we should leave things as is (a workaround adds unnecessary complexity).

**Comment by vpavic** [comment]
[UNDERTOW-1111](https://issues.jboss.org/browse/UNDERTOW-1111) has been fixed in undertow-io/undertow@7ee0452. The fix will be available in Undertow releases `1.4.17.Final` and `1.3.31.Final`.

With that, we can close this issue. Thanks @stevestorey for bringing this up.
